internal/rbd/controllerserver.go

```diff
@@ -831,6 +831,51 @@ func (cs *ControllerServer) DeleteVolume(
 	}
 	defer cs.VolumeLocks.Release(rbdVol.RequestName)
 
+	return cleanupRBDImage(ctx, rbdVol, cr)
+}
+
+// cleanupRBDImage removes the rbd image and OMAP metadata associated with it.
+func cleanupRBDImage(ctx context.Context,
+	rbdVol *rbdVolume, cr *util.Credentials) (*csi.DeleteVolumeResponse, error) {
+	mirroringInfo, err := rbdVol.getImageMirroringInfo()
```

Comment:
(question) So this would return no error when image mirroring is not enabled? Got till the go-ceph call, but wanted to validate nevertheless. And if image mirroring is not enabled the mirroringInfo  struct will have the right values.

Reply:
Yes, correct. it won't return an error message, the struct will be filled with the right values.